src/App.tsx

```diff
@@ -1,61 +1,104 @@
 import './App.scss';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { ChangeEvent, FormEvent, useState } from 'react';
+
+function getUserById(userId) {
+  return usersFromServer.find(user => user.id === userId) || null;
+}
+
+export const todos = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
 
 export const App = () => {
+  const [selectedUserId, setSelectedUserId] = useState(0);
+  const [text, setText] = useState('');
+  const [titleError, setTitleError] = useState(false);
+  const [userError, setUserError] = useState(false);
+  const [newTodos, setNewTodos] = useState(todos);
+
+  const handleUserChange = event => {
+    setSelectedUserId(Number(event.target.value));
+    if (userError) {
+      setUserError(false);
+    }
+  };
+
+  const handleTitleChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setText(event.target.value);
+  };
+
+  const handleSubmit = (event: FormEvent) => {
+    event.preventDefault();
+
+    if (!text) {
+      setTitleError(true);
+    }
+
+    if (!selectedUserId) {
+      setUserError(true);
+    }
+
+    if (text && selectedUserId) {
+      const newTodo = {
+        id: newTodos.length + 1,
+        title: text,
+        completed: false,
+        userId: selectedUserId,
+        user: usersFromServer.find(user => user.id === selectedUserId),
+      };
+
+      setNewTodos([...newTodos, newTodo]);
+      setText('');
+      setSelectedUserId(0);
+    }
```

Comment:
After successfully adding a new todo, the error states (`titleError` and `userError`) should be reset to `false`. This ensures that error messages do not persist after a successful submission.